### ISSUE TITLE
Reusing schema results in inconsistent autoValue behavoir

### ISSUE THREAD

**Issue by jschlieber** [aldeed/meteor-collection2-core#23]
I defined a schema to automatically create `createdAt` and `updatedAt` fields:

```js
import SimpleSchema from 'simpl-schema'

const timestampSchema = new SimpleSchema({
  'createdAt': {
    type: Date,
    autoValue () {
      if (this.isInsert) {
        return new Date()
      } else if (this.isUpsert) {
        return { $setOnInsert: new Date() }
      } else {
        this.unset()  // Prevent user from supplying their own value
      }
    }
  },
  'updatedAt': {
    type: Date,
    optional: true,
    autoValue () {
      console.log('updatedAt auto value...')
      if (this.isUpdate) {
        console.log('isUpdate...')
        return new Date()
      }
    }
  }
})

export default timestampSchema
```
I also have several collections where I attach schemas in the exact same way, like:

```js
import { Mongo } from 'meteor/mongo'
import SimpleSchema from 'simpl-schema'
import timestampSchema from './schemas/timestampSchema'

export const Collection = new Mongo.Collection('collection-name')

const schema = new SimpleSchema({
  '_id': {
    type: String,
    optional: true
  },
 ...
})

Collection.attachSchema(schema)
Collection.attachSchema(timestampSchema)
```

The collection `schema`s themselves don't add any `autoValue`s. There's really no difference between the collections other then their names and their field names. There are no other fields named `createdAt` or `updatedAt`. Now, when running updates on documents of any of the collections, the following log output appears
```
updatedAt auto value...
isUpdate...
```
as expected. But, for some reason, only documents of certain collections will get an `updatedAt` field. It's always the same collections where it will work and where it won't. I know this sounds very strange and if I saw such an issue I would immediately think there must be some difference that I missed. But how can it be, that the above log output appears for every update but there is no `updatedAt` field present in the updated document for specific collections? And yes, I checked, the updates themselves did work correctly.

Any help or ideas would be greatly appreciated. Thank you.

**Comment by jschlieber** [comment]
Heureka! I found a way to make it work for all collections. Found out this might actually be related to #8. But there is no Error thrown. Let's say `doc` is an object with `Object.getPrototypeOf(doc)` equals `null` and it is valid by the schema of the `Collection`. When I do an update
```js
Collection.update({
  _id: doc._id
}, {
  $set: doc
})
```
then no matter what any `autoValue` function may return (and they do all run), the result will not appear in the updated document. But if I do the update this way:
```js
Collection.update({
  _id: doc._id
}, {
  $set: {
    ...doc
  }
})
```
where `Object.getPrototypeOf({ ...doc })` equals `{}` it works and the `autoValue` fields will be there as expected.
So, it turns out it is not a problem of schema definitions and their attachments to the collections, but it's a problem of how update methods are called and it seems collection2 doesn't like 'prototype-less' objects.
